tools/stats/monitor.py

```
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import datetime
+import importlib.util
+import json
+import subprocess
+import sys
+import time
+from typing import Any, Dict, List
+
+
+def pip_install(package_name: str) -> None:
+    if importlib.util.find_spec(package_name) is None:
+        output = subprocess.run(
+            [sys.executable, "-m", "pip", "install", package_name],
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+        )
+        if output.returncode != 0:
+            print(output.stderr.decode("utf-8"))
+            print(output.stdout.decode("utf-8"))
+    # when running this script in the background, pip installs the packages to
+    # a location that is not in a path python uses to look for modules, so we
+    # manually add the location to the path
+    location = subprocess.run(
+        [sys.executable, "-m", "pip", "show", package_name],
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+    )
+    for s in location.stdout.decode("utf-8").splitlines():
+        if "Location:" in s:
+            sys.path.append(s.split(" ")[1])
+
+
+def main() -> None:
+    pip_install("psutil")
+    pip_install("pynvml")
+    import psutil  # type: ignore[import]
+    import pynvml  # type: ignore[import]
+
+    def get_processes_running_python_tests() -> List[Any]:
+        python_processes = []
+        for process in psutil.process_iter():
+            try:
+                if "python" in process.name() and process.cmdline():
+                    python_processes.append(process)
+            except Exception:
```

Can we use a more specific exception type here and don't catch all Exception?